### PR TITLE
Log IT error for non expected hits (#2330)

### PR DESCRIPTION
**Pull request by achave11**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `DCP/1` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make requirements_update` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

Primary reviewer (before pushing merge commit)

- [ ] Checked `reindex` label and `[r]` prefix of commit title
- [ ] Rebased and squashed branch
- [ ] Sanity-checked history
- [ ] Build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [ ] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [ ] Added PR reference to merge commit
- [ ] Moved linked issue to Merged
- [ ] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

**Review by danielsotirhos — APPROVED** [review]
Approved.

**Comment by hannes-ucsc** [comment]
Since I have taken over #2330 (because I have the fix), you should create a separate issue for this, @achave11. Be sure to link the two issues via comment.

File: test/integration_test.py [hunk 1]
```diff
@@ -559,11 +559,14 @@ def _assert_catalog_complete(self,
 
     def _assert_catalog_empty(self, catalog: CatalogName):
         with self.subTest('catalog_empty', catalog=catalog):
-            hit_counts = {
-                entity_type: len(self._get_entities(catalog, entity_type))
+            entities = {
+                entity_type: self._get_entities(catalog, entity_type)
                 for entity_type in self.entity_types
             }
+            hit_counts = {e: len(h) for e, h in entities.items()}
```
Comment (hannes-ucsc) [inline test/integration_test.py:566]: `entity_type` and `hits` respectively.

File: test/integration_test.py [hunk 2]
```diff
@@ -559,11 +559,14 @@ def _assert_catalog_complete(self,
...
             log.info('Hit counts are %r', hit_counts)
+            if any(hit_counts.values()):
```
Comment (hannes-ucsc) [inline test/integration_test.py:568]: This is essentially the worst possible way of doing this. `any(hit_counts.values())` is the same as `sum(hit_counts.values())`. Why do essentially the same check twice (inefficient) but in two different ways (confusing)? Exploit that assertEqual can be passed a message.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
And see my comment above on creating new issue.

**Comment by achave11** [comment]
@hannes-ucsc this branch can be deleted.

**Comment by hannes-ucsc** [comment]
> this branch can be deleted

Is there a replacement PR?

**Comment by hannes-ucsc** [comment]
Never mind. In the future, feel free to close and link yourself, as I will do now,.

**Comment by hannes-ucsc** [comment]
Continued in #2385.

**Pull request closed without merging**